rootfs/scheduler/k8s.py

```diff
@@ -502,33 +536,53 @@ def _get_service(self, name, namespace):
 
         return response
 
-    def _create_service(self, name, app_name, app_type, data={}):
-        actual_pod = {}
-        for _ in xrange(300):
-            status, json_data, reason = self._get_pods(app_name)
-            parsed_json = json.loads(json_data)
-            for pod in parsed_json['items']:
-                if('generateName' in pod['metadata'] and
-                   pod['metadata']['generateName'] == name + '-'):
-                    actual_pod = pod
-                    break
-
-            if actual_pod and actual_pod['status']['phase'] == 'Running':
-                break
-
-            time.sleep(1)
+    def _create_secret(self, namespace):
+        secretId, secretKey = '', ''
+        with open("/var/run/secrets/deis/minio/user/access-key-id") as the_file:
+            secretId = the_file.read()
+        with open("/var/run/secrets/deis/minio/user/access-secret-key") as the_file:
+            secretKey = the_file.read()
+        Key, Id = base64.b64encode(secretKey), base64.b64encode(secretId)
+        l = {
+            "version": self.apiversion,
+            "id": namespace,
+            "secretId": Id,
+            "secretKey": Key,
+            }
+        template = json.loads(string.Template(SECRET_TEMPLATE).substitute(l))
+        url = self._api("/namespaces/{}/secrets", namespace)
+        resp = self.session.post(url, json=template)
+        if unhealthy(resp.status_code):
+            error(resp, 'failed to create secret in Namespace "{}"', namespace)
 
-        container_id = actual_pod['status']['containerStatuses'][0]['containerID'].split("//")[1]
+    def _create_service(self, name, app_name, app_type, data={}):
+        # actual_pod = {}
+        # for _ in xrange(300):
+        #     status, json_data, reason = self._get_pods(app_name)
+        #     parsed_json = json.loads(json_data)
+        #     for pod in parsed_json['items']:
+        #         if('generateName' in pod['metadata'] and
+        #            pod['metadata']['generateName'] == name + '-'):
+        #             actual_pod = pod
+        #             break
+        #
+        #     if actual_pod and actual_pod['status']['phase'] == 'Running':
+        #         break
+        #
+        #     time.sleep(1)
+        #
+        # container_id = actual_pod['status']['containerStatuses'][0]['containerID'].split("//")[1]
         # ip = actual_pod['status']['hostIP']
         # TODO: more robust way of determining the first exposed port--this will only work on
         # the node where this deis/workflow pod is running.
         # Find the first exposed port by inspecting the Docker container
-        docker_cli = Client(version="auto")
```

Comment:
after speaking with @smothiki, this seemed to be necessary if we wanted to switch the deployment order from RC -> Service to Secret -> Service -> RC. We've talked it over and gonna change it a bit so the priority is Secret -> RC -> Service so we can still do port interpolation this way for the service.